astropy/io/fits/connect.py

```diff
@@ -392,3 +446,161 @@ def write_table_fits(input, output, overwrite=False):
+
+
+def _fortran_to_python_format(tdisp):
+    """
+    Turn the TDISPn fortran format pieces into a final Python format string.
+    See the format_type definitions above the TDISP_FMT_DICT. If codes is
+    changed to take advantage of the exponential specification, will need to
+    add it as another input parameter.
+
+    Parameters
+    ----------
+    tdisp: str
+        TDISPn FITS Header keyword.  Used to specify display formatting.
+
+    Returns
+    -------
+    format_string: str
+        The TDISPn keyword string translated into a Python format string.
+    """
+    format_type, width, precision, exponential = _parse_tdisp_format(tdisp)
+
+    try:
+        fmt = TDISP_FMT_DICT[format_type]
+        return fmt.format(width=width, precision=precision)
+
+    except KeyError:
+        raise VerifyError('Format {} is not recognized.'.format(format_type))
+
+
+def python_to_tdisp(format_string):
+    """
+    Turn the Python format string to a TDISP FITS compliant format string. Not
+    all formats convert. these will cause a Warning and return None.
+
+    Parameters
+    ----------
+    format_string: str
+        TDISPn FITS Header keyword.  Used to specify display formatting.
+
+    Returns
+    -------
+    tdsip_string: str
+        The TDISPn keyword string translated into a Python format string.
+    """
+
+    fmt_to_tdisp = {'a': 'A', 's': 'A', 'd': 'I', 'b': 'B', 'o': 'O', 'x': 'Z',
+                    'X': 'Z', 'f': 'F', 'F': 'F', 'g': 'G', 'G': 'G', 'e': 'E',
+                    'E': 'E'}
+
+    if format_string is None or format_string == "":
+        return None
+
+    # Strip out extra format characters that aren't a type or a width/precision
+    if format_string[0] == '{' and format_string != "{}":
+        fmt_str = format_string.lstrip("{:").rstrip('}')
+    elif format_string[0] == '%':
+        fmt_str = format_string.lstrip("%")
+    else:
+        warnings.warn('Format {} cannot be mapped to the accepted '
+                      'TDISPn keyword values.  Format will not be '
+                      'moved into TDISPn keyword.'.format(format_string),
+                      AstropyUserWarning)
+        return None
+
+    precision, sep = '', ''
+
+    # Character format, only translate right aligned, and don't take zero fills
+    if fmt_str[-1].isdigit() and fmt_str[0] == '>' and fmt_str[1] != '0':
+        ftype = fmt_to_tdisp['a']
+        width = fmt_str[1:]
+
+    elif fmt_str[-1] == 's' and fmt_str != 's':
+        ftype = fmt_to_tdisp['a']
+        width = fmt_str[:-1].lstrip('0')
+
+    # Number formats, don't take zero fills
+    elif fmt_str[-1].isalpha() and len(fmt_str) > 1 and fmt_str[0] != '0':
+        ftype = fmt_to_tdisp[fmt_str[-1]]
+        fmt_str = fmt_str[:-1]
+
+        # If format has a "." split out the width and precision
+        if '.' in fmt_str:
+            width, precision = fmt_str.split('.')
+            sep = '.'
+            if width == "":
+                width = precision
```

Comment:
If the width is not specified, it may be better to use some default width instead ? E.g. the maximum width for the given type, maybe inspired from ASCII_DEFAULT_WIDTHS:
 https://github.com/astropy/astropy/blob/8ad19c68ddbc2ac33eaf90c1ff988991bbdcc745/astropy/io/fits/column.py#L78-L79

Reply:
Alternatively, one could match this with python - which also does not require a width for its formats.

Reply:
I feel pretty ambivalent about adding a default width versus using the precision.  Should I literally use the  ASCII_DEFAULT_WIDTHS values for this @saimn? @mhvk, we have to have some width set, it's required by TDISP. Unless your suggestion is to not move those values into TDISP at all?

Reply:
@SaraOgaz - as I fear too often is the case, I didn't look carefully enough before commenting...  I think what I'd do is `width = precision + (default_width - default_precision)` - that will take care of `±0.` before the "precision" part and `E+000` after (for exponential formats).

Reply:
@mhvk, what would default_precision be in this case?

Reply:
By `default_width` and `default_precision`, I meant the two items in the tuples linked by @saimn, but I have to admit that I didn't actually check that that is what those are supposed to be. Fortunately, I see to be right: looking at the file, I see that a preceding comment:
```
# For each ASCII table format code, provides a default width (and decimal
# precision) for when one isn't given explicitly in the column format
ASCII_DEFAULT_WIDTHS = {'A': (1, 0), 'I': (10, 0), 'J': (15, 0),
                        'E': (15, 7), 'F': (16, 7), 'D': (25, 17)}
```

Reply:
Ahh, thanks, missed that bit in the comment.

Reply:
Yes it is used for the ASCII table formats, for TFORM. And looking more in detail it looks very similar to the parsing done here for TDISP - except that I think ASCII table support fewer formats:

https://github.com/astropy/astropy/blob/8ad19c68ddbc2ac33eaf90c1ff988991bbdcc745/astropy/io/fits/column.py#L2115-L2124

Maybe there are other useful things to take from there ? I don't know well enough this code and won't have time to look in detail, so it's just an idea, if you want to have a look @SaraOgaz .At least having the same default with and precision seems sensible.

Reply:
@saimn (@mhvk), yeah I used that heavily as a guide for the TDISP version.  I don't think there's quite enough overlap to use the same function for both these tasks, especially because TDISP needs the additional parsing for exponential. So not sure what else I would take from this that's not already in  `_parse_tdisp_format`...